orchagent/errororch.cpp

```diff
@@ -0,0 +1,558 @@
+         if(entryFound == true)
+         {
+             removeEntry = true;
+         }
+     }
+     else
+     {
+         if(strOp == "create")
+         {
+             addEntry = true;
+         }
+         else if(strOp == "remove")
+         {
+             if(entryFound == true)
+             {
+                 removeEntry = true;
+             }
+             else
+             {
+                 addEntry = true;
+             }
+         }
+         else if(strOp == "set")
+         {
+             if(entryFound == true)
+             {
+                 updateEntry = true;
+             }
+             else
+             {
+                 addEntry = true;
+             }
+         }
+     }
+     SWSS_LOG_DEBUG("key %s operation %s RC %s Entry in DB %d add:remove:update %d:%d:%d ",
+             key.c_str(), strOp.c_str(), strRc.c_str(),
+             entryFound, addEntry, removeEntry, updateEntry);
+     if(addEntry == true || updateEntry == true)
+     {
+         /* Create/Update the database entry */
+         table->set(key, values);
+         SWSS_LOG_INFO("Publish %s(ok) to error db", key.c_str());
+     }
+     else if (removeEntry == true)
+     {
+         /* Remove the entry from database */
+         table->del(key);
+         SWSS_LOG_INFO("Removed %s(ok) from error db", key.c_str());
+     }
+
+     return;
+ }
+
+/* The following function is used by other Orch Agents to add the entry into *
+ * error database and optionally send a notification to the applications     */
+void ErrorOrch::addErrorEntry(sai_object_type_t object_type,
+        std::vector<FieldValueTuple> &appValues, uint32_t flags)
+{
+    SWSS_LOG_ENTER();
+
+    std::shared_ptr<Table> table;
+
+    /*
+     * The following steps are performed:
+     * Find out the table object based on the object type
+     * Update DB if the notification is about failure
+     * Send notification to error listener class if the flag is set
+     */
+
+    std::string str_object_type = sai_serialize_object_type(object_type);
+    SWSS_LOG_DEBUG("Field values received for %s: ", str_object_type.c_str());
+    for (size_t i = 0; i < appValues.size(); i++)
+    {
+        SWSS_LOG_DEBUG("%s -> %s", fvField(appValues[i]).c_str(),
+                fvValue(appValues[i]).c_str());
+    }
+
+    if(m_ObjTableMap.find(str_object_type) == m_ObjTableMap.end())
+    {
+        SWSS_LOG_INFO("EHF does not support SAI object type %s",
+                str_object_type.c_str());
+        return;
+    }
+
+    string tableName = m_ObjTableMap[str_object_type];
+    if(m_TableOrchMap.find(tableName) == m_TableOrchMap.end())
+    {
+        SWSS_LOG_INFO("EHF does not support error handling for %s",
+                tableName.c_str());
+        return;
+    }
+
+    string errKey;
+    auto dbValues = appValues;
+
+    /* Remove key from FV as it's passed explicitly while updating DB */
+    auto iu = dbValues.begin();
+    while (iu != dbValues.end())
```

Comment:
Same comments as above (you also might consider making this loop its own method since it's used in a few different spots).

Reply:
Added a new method to extract the specified field